apps/addons/buttons.py

```diff
@@ -233,11 +234,13 @@ def __init__(self, text, url, os=None, file=None):
 
 # Cache it for a year.
 @cache_page(60 * 60 * 24 * 365)
+@non_atomic_requests
```

Comment:
from a look at the [non_atomic_requests code](https://github.com/django/django/blob/1.6.11/django/db/transaction.py#L387-L389), the ordering isn't so important. It just needs a chance to add an attribute to the view.